### ISSUE TITLE
bug in BYTSCL (coming from internal MinMax() and affecting MIN() & MAX())

### ISSUE THREAD

**Issue by alaingdl** [gnudatalanguage/gdl#1928]
I discovered by chance this bug because in the image I read back we have a NaN at a corner ...

```
a=dist(128) & tvscl, a & print, min(bytscl(a)), max(bytscl(a))
   0 255
GDL> a[30,0]=!values.f_nan & tvscl, a & print, min(bytscl(a)), max(bytscl(a))
   0 255
GDL> a[0,0]=!values.f_nan & tvscl, a & print, min(bytscl(a)), max(bytscl(a))
  0     0
```

**Comment by alaingdl** [comment]
After some tests: 
1. only first position is a problem : at  [0,0] 
2. it seems to be related only to +- NaN,
3. but it is OK for +- Inf

```
GDL> a=dist(4) & a[0,0]=-!values.f_infinity & tvscl, a & print, min(bytscl(a)), max(bytscl(a))
   0 255
```

And I go directly to line 7195 in `basic_fun.cpp` : 
```
   dRes->MinMax(&minEl, &maxEl, NULL, NULL, omitNaN);
```
but I don't understand why the occurrence at first place make a difference 
I suspect a trick inside MinMax()

**Comment by alaingdl** [comment]
At first place

```
IDL> a=dist(4) & a[0,0]=!values.f_nan & print, bytscl(a)
   0   0 140   0
   0  57 173  57
 140 173 255 173
   0  57 173  57
GDL> a=dist(4) & a[0,0]=!values.f_nan & print, bytscl(a)
   0   0   0   0
   0   0   0   0
   0   0   0   0
   0   0   0   0
```

At other place
```
IDL> a=dist(4) & a[0,3]=!values.f_nan & print, bytscl(a)
   0  90 181  90
  90 127 202 127
 181 202 255 202
   0 127 202 127
GDL> a=dist(4) & a[0,3]=!values.f_nan & print, bytscl(a,/na)
   0  90 181  90
  90 127 202 127
 181 202 255 202
   0 127 202 127
```

**Comment by alaingdl** [comment]
OK, as I suspected, the problem *is* inside MinMax() 

```
GDL> a=dist(128) & a[30,0]=!values.f_nan & print, min(a), max(a)
      0.00000      90.5097
GDL> a=dist(128) & a[0,0]=!values.f_nan & print, min(a), max(a)
          NaN          NaN

IDL> a=dist(128) & a[30,0]=!values.f_nan & print, min(a), max(a)
% Compiled module: DIST.
      0.00000      90.5097
IDL> a=dist(128) & a[0,0]=!values.f_nan & print, min(a), max(a)
      1.00000      90.5097

```

**Comment by GillesDuvert** [comment]
This IS NOT A BUG. :smile: 
(but this is of course a bug, @alaingdl )!

this is not a bug because:
```
GDL> print,!version.release
8.2
```
and
```
IDL> print,!version.release
8.4
IDL> a=dist(4) & a[0,0]=!values.f_nan & print, bytscl(a)
% Compiled module: DIST.
   0   0   0   0
   0   0   0   0
   0   0   0   0
   0   0   0   0
```
this is related to #1353 . Nobody yet has forced /NAN in the code. This would have been sooo easy.

**Comment by GillesDuvert** [comment]
closed by #1946
